version: '3.1'

services:
  mysql_dev:
    image: mysql:5.7.35
    
    command: --default-authentication-plugin=mysql_native_password
    
    restart: always

    volumes:
      - ./services/mysql_dev/data:/var/lib/mysql
      - ./services/mysql_dev/log:/var/log/mysql
      - ./services/mysql_dev/config:/etc/mysql
      - ./files:/files

    env_file:
      - .prod.env
    
    ports:
      - 3310:3306
  
  phpmyadmin:
    image: phpmyadmin/phpmyadmin:4.9.7

    restart: always
    
    ports:
      - 3311:80
    
    environment:
      - PMA_ARBITRARY=0
      - PMA_HOST=mysql_dev
      - PMA_PORT=3306
  
  redis:
    image: "redis:alpine"
    
    hostname: redis

    volumes:
      - ./services/redis/data:/data
    ports:
      - 6380:6379
    restart: always

  mongodb:
    
    image: mongo:latest

    env_file:
      - .prod.env

    ports:
      - 27020:27017

    volumes:
      - ./services/mongodb/database:/data/db
    
    restart: always
  
  pulsar:
    image: apachepulsar/pulsar:2.8.1

    volumes:
      - ./services/pulsar/pulsardata:/pulsar/data
    
    restart: always
    
    ports:
      - 6655:6650
      - 8080:8080
    
    command: "bin/pulsar standalone -nss"

    restart: always
  
  notebook:
    container_name: python_notebook
    ports:
      - 9090:8888
    volumes:
      - ./services/jupyter/notebooks:/notebooks
      - /var/www/html/files:/files

    build: ./services/jupyter
    
    network_mode: "host"